fmriprep/interfaces/bids.py

```diff
@@ -0,0 +1,62 @@
+"""BIDS-related interfaces."""
+
+from pathlib import Path
+
+from bids.utils import listify
+from nipype.interfaces.base import (
+    DynamicTraitedSpec,
+    SimpleInterface,
+    TraitedSpec,
+    isdefined,
+    traits,
+)
+from nipype.interfaces.io import add_traits
+from nipype.interfaces.utility.base import _ravel
+
+from ..utils.bids import _find_nearest_path
+
+
+class _BIDSURIInputSpec(DynamicTraitedSpec):
+    dataset_links = traits.Dict(mandatory=True, desc='Dataset links')
+    out_dir = traits.Str(mandatory=True, desc='Output directory')
+
+
+class _BIDSURIOutputSpec(TraitedSpec):
+    out = traits.List(
+        traits.Str,
+        desc='BIDS URI(s) for file',
+    )
+
+
+class BIDSURI(SimpleInterface):
+    """Convert input filenames to BIDS URIs, based on links in the dataset.
+
+    This interface can combine multiple lists of inputs.
+    """
+
+    input_spec = _BIDSURIInputSpec
+    output_spec = _BIDSURIOutputSpec
+
+    def __init__(self, numinputs=0, **inputs):
+        super().__init__(**inputs)
+        self._numinputs = numinputs
+        if numinputs >= 1:
+            input_names = [f'in{i + 1}' for i in range(numinputs)]
+        else:
+            input_names = []
+        add_traits(self.inputs, input_names)
```

Comment:
Not sure if I fully understand, but seems a bit over complicated recreating Nipype's Merge behavior here when we could just stick a Merge node beforehand to aggregate / ravel lists of files if needed

Reply:
It could be overcomplicated. I was just thinking of how things work in XCP-D (which currently uses functions for this, rather than an interface). I have to build several different sets of Sources depending on the output file, so having a Merge before each and every case can result in some pretty overpopulated workflows. I figured that  passing the output from one or more BIDSURI nodes directly into a later one would be easier most of the time.

Reply:
Fair enough - I prefer simplicity when designing interfaces but this is more of a nitpick